### PR TITLE
autoscaling: update auto scaling API

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Related to #2621. After #2673, there are still some problems which are existed operator or TiDB cloud implementation.

### What is changed and how it works?

This PR updates the API to make it easier.

### Check List

Tests

- Unit test

Code changes

- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/pingcap/pd/blob/master/docs/development.md#updating-api-documentation))

### CODE REVIEW

File: tests/autoscaling/autoscaling_test.go [hunk 1]
```diff
@@ -89,35 +88,20 @@ func (s *apiTestSuite) TestAPI(c *C) {
...
-            "count":2
-        }
-    ],
     "resources":[
         {
             "resource_type":"resource_a",
             "cpu":1,
             "memory":8,
-            "storage":1000
+            "storage":1000,
```
Comment (howardlau1999) [inline tests/autoscaling/autoscaling_test.go:96]: The unit here is not the same as the definition

Comment (rleungx) [reply tests/autoscaling/autoscaling_test.go:96]: It is only a test for testing the function of serialization.

**Comment by nolouch** [comment]
LGTM

**Pull request merged**